pkg/cloudprovider/aws/suite_test.go

```diff
@@ -220,6 +223,106 @@ var _ = Describe("Allocation", func() {
 			Expect(scheduled1.Spec.NodeName).NotTo(Equal(scheduled2.Spec.NodeName))
 			Expect(fakeEC2API.CalledWithCreateFleetInput).To(HaveLen(2))
 		})
+		It("should launch instances for Nvidia GPU resource requests", func() {
+			// Setup
+			pod1 := test.PendingPodWith(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{resources.NvidiaGPU: resource.MustParse("1")},
+					Limits:   v1.ResourceList{resources.NvidiaGPU: resource.MustParse("1")},
+				},
+			})
+			// Should pack onto same instance
+			pod2 := test.PendingPodWith(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{resources.NvidiaGPU: resource.MustParse("2")},
+					Limits:   v1.ResourceList{resources.NvidiaGPU: resource.MustParse("2")},
+				},
+			})
+			// Should pack onto a separate instance
+			pod3 := test.PendingPodWith(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Requests: v1.ResourceList{resources.NvidiaGPU: resource.MustParse("4")},
+					Limits:   v1.ResourceList{resources.NvidiaGPU: resource.MustParse("4")},
+				},
+			})
+			ExpectCreatedWithStatus(env.Client, pod1, pod2, pod3)
+			ExpectCreated(env.Client, provisioner)
```

Comment:
Is it possible to have different functionality based on when pods are created in relation to when the provisioner is created? E.g. if pods are created after the provisioner vs before the provisioner, will different actions be taken? This could be a source of race conditions in other tests as well.

Reply:
This is 100% the source of our race conditions. The pods can come in at any time, which can result in different packings. Likely we need to do something like 

1. Create Pods
2. Sync Informer Caches
3. Execute a single reconciliation loop

But this requires significant changes to our testing.

Reply:
Ginkgo has a couple of race-condition settings that we've been using. Do you think we need more robust race-condition checking? If we had more deterministic test functions, this could solve our problems. Right now we're at the mercy of the environment.

Reply:
Ginkgo's race isn't good enough it seems. Let's tackle this as a separate effort.